sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/models/QueuesGetPropertiesHeaders.java

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.storage.queue.implementation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.annotation.HeaderCollection;
+import com.azure.core.util.DateTimeRfc1123;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+/** The QueuesGetPropertiesHeaders model. */
+@JacksonXmlRootElement(localName = "null")
+@Fluent
+public final class QueuesGetPropertiesHeaders {
+    /*
+     * The x-ms-version property.
+     */
+    @JsonProperty(value = "x-ms-version")
+    private String xMsVersion;
+
+    /*
+     * The x-ms-approximate-messages-count property.
+     */
+    @JsonProperty(value = "x-ms-approximate-messages-count")
+    private Integer xMsApproximateMessagesCount;
+
+    /*
+     * The x-ms-meta- property.
+     */
+    @HeaderCollection(value = "x-ms-meta-")
```

Comment:
HeaderCollection still needs to be resolved by autorest java. Right now it generates JsonProperty instead of HeaderCollection